tools/shared.py

```diff
@@ -1384,6 +1384,12 @@ def remove_quotes(arg):
 
   @staticmethod
   def get_building_env(native=False, doublequote_commands=False):
+    # CMake does not like 'CC=python emcc.py', CMAKE_C_COMPILER becomes just 'python' without emcc
+    def shellpath(pypath):
+      ext = '.bat' if WINDOWS else ''
+      if pypath.endswith('.py'):
+        pypath = pypath[:-3]
```

Comment:
why remove the suffix?

Reply:
CMake splits `python emcc.py` into `python` and thinks `python` is the C compiler. I think this forces us to roll back to `emcc` from `python emcc.py` at least when setting environment variables.